frontend/src/ui/InitialConsent.tsx

```diff
@@ -0,0 +1,42 @@
+import { useRef } from "react";
+import CONFIG from "../config";
+import { currentRef, useTranslatedConfig } from "../util";
+import { Modal, ModalHandle } from "./Modal";
+import { Button } from "./Button";
+import { TextBlock } from "./Blocks/Text";
+
+
+export const InitialConsent: React.FC = () => {
+    const modalRef = useRef<ModalHandle>(null);
+    const userConsent = localStorage.getItem("userConsent") ?? "";
+    const { title, text, button } = CONFIG.initialConsent;
+
+    // Since this doesn't store any critical information, a simple, insecure hash
+    // should suffice.
+    // Source: https://gist.github.com/jlevy/c246006675becc446360a798e2b2d781
+    const simpleHash = (str: string) => {
+        let hash = 0;
+        for (let i = 0; i < str.length; i++) {
+            hash = ((hash << 5) - hash + str.charCodeAt(i)) | 0;
+        }
+        return (hash >>> 0).toString(36);
+    };
```

Comment:
I like that you thought of performance, this is something too many programmers completely ignore. But in performance optimization, it's equally important to know what *not* to optimize. Because in the end, it's not about finding optimization opportunities; every decently sized project has tons of those. It's about picking the ones with the most improvement per time investment. Again, the limiting factor in performance optimization is not the number of places one could optimize, but the programmers time.

And in this particular case: calculating the SHA256 hash of a 8,5GB file via `sha256sum` takes 18s on my PC. That's roughly 2µs per KB of data. Even with startup/teardown overhead, it's still completely negligible in our situation where we only hash once, with data in the order of magnitude of 1KB.